codec-http2/src/main/java/io/netty/handler/codec/http2/Http2RemoteFlowController.java

```diff
@@ -21,6 +21,13 @@
  * endpoint.
  */
 public interface Http2RemoteFlowController extends Http2FlowController {
+    /**
+     * Get the {@link ChannelHandlerContext} for which to apply flow control on.
+     * <p>
+     * This is intended for us by {@link FlowControlled} implementations only. Use with caution.
+     * @return The {@link ChannelHandlerContext} for which to apply flow control on.
+     */
+    ChannelHandlerContext channelHandlerContext();
```

Comment:
This is not ideal but it was added to avoid creating another object (`ChannelFutureListener`) for each `FlowControlled` frame. The `FlowControlled` only uses this when an asynchronous error happens on the promise it holds.  In all other "synchronous" cases (or errors detected by the flow controller) `FlowControlled` is provided the `ChannelHandlerContext` in an argument from the flow controller.